diff --git a/python/sglang/srt/sampling/penaltylib/min_new_tokens.py b/python/sglang/srt/sampling/penaltylib/min_new_tokens.py
--- a/python/sglang/srt/sampling/penaltylib/min_new_tokens.py
+++ b/python/sglang/srt/sampling/penaltylib/min_new_tokens.py
@@ -25,15 +25,18 @@ class BatchedMinNewTokensPenalizer(_BatchedPenalizer):
         padded_stop_token_ids = torch.nn.utils.rnn.pad_sequence(
             sequences=[
                 torch.tensor(
-                    data=(
-                        list(
-                            (req.sampling_params.stop_token_ids or set())
-                            | (req.tokenizer.additional_stop_token_ids or set())
-                            | {req.tokenizer.eos_token_id}
+                    data=list(
+                        token_id
+                        for token_id in (
+                            (req.sampling_params.stop_token_ids or set())
+                            | (req.eos_token_ids or set())
+                            | (req.tokenizer.additional_stop_token_ids or set())
+                            | {req.tokenizer.eos_token_id}
                         )
+                        if token_id is not None
                     ),
                     dtype=torch.int64,
                     device=self.orchestrator.device,
                 )
                 for req in self.orchestrator.reqs()
             ],
